test/smartwallet/CustomSmartWallet.test.ts

```diff
@@ -0,0 +1,340 @@
+
+      const initialWorkerTokenBalance = await getTokenBalance(token, worker)
+      const initialSWalletTokenBalance = await getTokenBalance(token, smartWallet.address)
+      const initialNonce = await smartWallet.nonce()
+
+      const relayRequest = await createRequest({
+        data: recipientFunction,
+        to: recipient.address,
+        nonce: initialNonce.toString(),
+        relayHub: worker,
+        tokenContract: token.address,
+        from: senderAddress
+      }, relayData)
+      const { signature, suffixData } = signRequest(senderPrivateKey, relayRequest, chainId)
+
+      const result = await smartWallet.execute(domainSeparatorHash, suffixData, relayRequest.request, signature, { from: worker })
+
+      // @ts-ignore
+      assert(containsEvent(customLogic.abi, result.receipt.rawLogs, 'LogicCalled'), 'Should call custom logic')
+
+      const tknBalance = await getTokenBalance(token, worker)
+      const swTknBalance = await getTokenBalance(token, smartWallet.address)
+
+      assert.equal(tknBalance.sub(initialWorkerTokenBalance).toString(), new BN(1).toString(), 'Incorrect new worker token balance')
+      assert.equal(initialSWalletTokenBalance.sub(swTknBalance).toString(), new BN(1).toString(), 'Incorrect new smart wallet token balance')
+
+      assert.equal((await smartWallet.nonce()).toString(), initialNonce.add(new BN(1)).toString(), 'verifyAndCall should increment nonce')
+    })
+
+    it("should call function from custom logic with wallet's address", async () => {
+      // Init smart wallet and
+      const customLogic = await ProxyCustomLogic.new()
+      await smartWallet.initialize(senderAddress, customLogic.address, token.address, worker, '0', '400000', '0x')
+
+      const initialWorkerTokenBalance = await getTokenBalance(token, worker)
+      const initialSWalletTokenBalance = await getTokenBalance(token, smartWallet.address)
+
+      const initialNonce = await smartWallet.nonce()
+
+      const relayRequest = await createRequest({
+        data: recipientFunction,
+        to: recipient.address,
+        nonce: initialNonce.toString(),
+        relayHub: worker,
+        tokenContract: token.address,
+        from: senderAddress
+      }, relayData)
+      const reqData: EIP712TypedData = new TypedRequestData(chainId, smartWallet.address, relayRequest)
+      const sig = signTypedData_v4(senderPrivateKey, { data: reqData })
+      const suffixData = bufferToHex(TypedDataUtils.encodeData(reqData.primaryType, reqData.message, reqData.types).slice((1 + countParams) * 32))
+
+      const result = await smartWallet.execute(domainSeparatorHash, suffixData, relayRequest.request, sig, { from: worker })
+
+      // @ts-ignore
+      assert(containsEvent(customLogic.abi, result.receipt.rawLogs, 'LogicCalled'), 'Should call custom logic')
+
+      // @ts-ignore
+      const logs = await recipient.getPastEvents('TestForwarderMessage')
```

Comment:
Can you read the event from the logs of the transaction receipt? (you can call web3.eth.getTransactionReceipt(txHash) again, as I noticed sometimes the rawLogs from the receipt sent by Truffle does not contain all the logs). Im worried you could be reading an event emitted in a previous transaction